### PR TITLE
ta: pkcs11: fixes for key check value + default enable its support

### PR DESCRIPTION
**Pull request by etienne-lms**
This series fixes PKCS11 key check value implementation and default enable its support.
With these changes, PKCS11 key check value is now tested by OP-TEE CI tests and should succeed.

### CODE REVIEW

**Comment by etienne-lms** [comment]
I enable by `CFG_PKCS11_TA_CHECK_VALUE_ATTRIBUTE` by default as per the outcome of discussion thread https://github.com/OP-TEE/optee_os/pull/6550#discussion_r1430368998.

File: ta/pkcs11/src/pkcs11_attributes.c [hunk 1]
```diff
@@ -2897,7 +2897,7 @@ static enum pkcs11_rc compute_check_value_with_ecb(void *key, uint32_t key_size,
 	TEE_OperationHandle op = TEE_HANDLE_NULL;
 	TEE_ObjectHandle hkey = TEE_HANDLE_NULL;
 	TEE_Attribute attr = { };
-	uint8_t *buf = NULL;
+	uint8_t buf[TEE_AES_BLOCK_SIZE] = { };
 	size_t buf_size = 0;
```
Comment (jforissier) [inline ta/pkcs11/src/pkcs11_attributes.c:2901]: = sizeof(buf);

**Review by jforissier — COMMENTED** [review]
One comment for "ta: pkcs11: fix block size to compute AES key check value", then:

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by jforissier — COMMENTED** [review]
For "ta: pkcs11: use TEE_InitRefAttribute() to compute key check value":

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
For "ta: pkcs11: default enable key check value support":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
For "ta: pkcs11: key check value may not be applicable":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
Comment addressed:
```patch
-	size_t buf_size = 0;
+	size_t buf_size = sizeof(buf);
 
 	(...)
 
-	buf_size = sizeof(buf);
```

Review tags applied.

**Comment by etienne-lms** [comment]
Thanks

**Comment by maroueneboubakri** [comment]
@jforissier hope the PR is ready to be merged along with https://github.com/OP-TEE/optee_test/pull/719

**Pull request merged**